src/components/Collapsable/Collapsable.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { css } from 'glamor'
+
+import { sansSerifRegular14 } from '../Typography/styles'
+
+import colors from '../../theme/colors'
+import { mUp } from '../../theme/mediaQueries'
+import { useMediaQuery } from '../../lib/useMediaQuery'
+import { useBoundingClientRect } from '../../lib/useBoundingClientRect'
+
+const COLLAPSED_HEIGHT = {
+  mobile: 180,
+  desktop: 240
+}
+
+const collapsedBodyStyle = (mobile, desktop) => {
+  return css({
+    overflow: 'hidden',
+    maxHeight: `${mobile}px`,
+    [mUp]: {
+      maxHeight: `${desktop}px`
+    }
+  })
+}
+
+const styles = {
+  buttonContainer: css({
+    position: 'relative',
+    borderTop: `1px solid ${colors.divider}`,
+    '&::before': {
+      position: 'absolute',
+      display: 'block',
+      content: '""',
+      left: 0,
+      right: 0,
+      top: -61,
+      height: 60,
+      background: 'linear-gradient(0deg, rgba(255,255,255,1) 0%, rgba(255,255,255,0) 100%)'
+    }
+  }),
+  button: css({
+    ...sansSerifRegular14,
+    outline: 'none',
+    WebkitAppearance: 'none',
+    background: 'transparent',
+    border: 'none',
+    padding: '0',
+    display: 'block',
+    cursor: 'pointer',
+    color: colors.primary,
+    height: '32px',
+    lineHeight: '32px',
+    '@media (hover)': {
+      ':hover': {
+        color: colors.secondary
+      }
+    }
+  })
+}
+
+const Collapsable = ({ t, children, collapsable, height, threshold, style }) => {
+  /**
+   * Measuring the body size (height), so we can determine whether to collapse
+   * the body.
+   *
+   * bodyVisibility:
+   *   - 'indeterminate': We don't know yet whether to collapse the body or not.
+   *   - 'full': The body is collapsable but we're showing the full body.
+   *   - 'preview': The body is collapsed.
+   */
+
+  const [bodyVisibility, setBodyVisibility] = React.useState('indeterminate')
```

Comment:
we should immediately render the preview state in the infobox case
```suggestion
  const [bodyVisibility, setBodyVisibility] = React.useState(threshold ? 'indeterminate' : 'preview')
```